Sources/TokamakCore/Fiber/Layout/PaddingLayoutComputer.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 5/28/22.
+//
+
+import Foundation
+
+private extension EdgeInsets {
+  init(applying edges: Edge.Set, to insets: EdgeInsets) {
+    self.init(
+      top: edges.contains(.top) ? insets.top : 0,
+      leading: edges.contains(.leading) ? insets.leading : 0,
+      bottom: edges.contains(.bottom) ? insets.bottom : 0,
+      trailing: edges.contains(.trailing) ? insets.trailing : 0
+    )
+  }
+}
+
+/// A `LayoutComputer` that fits to its children then adds padding.
+final class PaddingLayoutComputer: LayoutComputer {
```

Comment:
I think I had a similar comment before, but got no clarity on this yet: are types conforming to `LayoutComputer` supposed to be classes or structs? If structs are fine, let's avoid classes without an explicit need for them, which helps us avoid slower allocations on the heap.

Reply:
…and if they should be classes, then the `LayoutComputer` protocol should inherit from `AnyObject` to ensure that.

Reply:
They should be structs now, I has changed that before merging the layout branch, but had made this PR before that change was made.